### ISSUE TITLE
Provide an extension point for generating arbitrary pages

### ISSUE THREAD

**Issue by IgnatBeresnev** [Kotlin/dokka#2771]
Currently, `SingleModuleGeneration` only supports creating pages from `Documentable` model using `DocumentableToPageTranslator` extensions.

However, a plugin may want to generate its own additional pages which are not created from documentables - for instance, by parsing self-defined `.md` documentation files. This would add the possibility of embedding tutorials/non-code documentation into the generated output.

`PageCreator` interface exists already, it's used as an extension point in multimodule plugin only for creating [all-modules page](https://kotlinlang.org/api/kotlinx.coroutines/), but it works by overrides `Generation`, which is not desirable for a small plugin.

`PageCreator` should be re-used and added as a base extension point, allowing plugins to produce self-defined pages.

**Comment by CLOVIS-AI** [comment]
### Example use case

The current markdown parser, as used to generate the module and package descriptions from external markdown files, has all features for basic documentation pages. It is also sometimes useful to write longer-form pages that describe non-source-node elements (e.g. tutorials). 

Having the ability to use the current markdown parser to generate additional pages would make Dokka a good fit to document libraries that are mostly Dokka + some small quantity of additional documentation, like Gradle setup, etc, that are not complicated enough to warrant the setup of a full on mkdoc or similar, but would clutter the documentation if placed in the module header.

**Comment by hfhbd** [comment]
Yeah, that would be really cool! I don't know if this is required for https://github.com/Kotlin/dokka/issues/2769 too or in scope of `Documentable`

**Comment by IgnatBeresnev** [comment]
Related: #1965
